src/sentry/rules/conditions/event_frequency.py

```diff
@@ -190,7 +190,7 @@ def passes(self, event: GroupEvent, state: EventState) -> bool:
         return current_value > value
 
     def passes_activity_frequency(
-        self, activity: ConditionActivity, buckets: dict[datetime, int | float]
```

Comment:
`int | float` is redundant in the python typing system -- that's just `float`

might be enough to allow the callers properly without resorting to `Mapping` everywhere

Reply:
Hmm are you suggesting changing these to `dict[datetime, float]`? Mypy doesn't seem particular happy about that when a `dict[datetime, int]` is passed: 
```
src/sentry/rules/history/preview.py:474: error: Argument 2 to "passes_activity_frequency" of "BaseEventFrequencyCondition" has incompatible type "dict[datetime, int]"; expected "dict[datetime, float]"
```

And changing the source data type seems wrong.

Reply:
actually -- I think the type should be `dict[datetime, int]` -- I can't see any place that `float`s are passed into here